src/circuit.rs

```diff
@@ -397,12 +397,20 @@ impl plonk::Circuit<pallas::Base> for Circuit {
 
         // Merkle path validity check.
         let anchor = {
+            let path = self.path.map(|typed_path| {
+                // TODO: Replace with array::map once MSRV is 1.55.0.
+                let mut inner_path = [pallas::Base::zero(); MERKLE_DEPTH_ORCHARD];
+                for (a, b) in inner_path.iter_mut().zip(typed_path.iter()) {
+                    *a = b.inner();
+                }
+                inner_path
```

Comment:
Use `gen_const_array` here.

Reply:
I don't understand this comment; this isn't creating an array of a single constant value?

Reply:
I guess you'd use indexed access into typed_path to obtain the values? I cordially dislike indexed access compared to operating on iterators like this.

Reply:
The thing with iterators like this is you need to be careful to ensure the lengths match, or you get short-cutting. We were careful in `gen_const_array` to use constants for lengths and indexing in the closure so that we'd get panics instead of silent truncation if the source array was shorter than the destination.

Reply:
In any case, once we can use `array::map`, all this goes away.